resources/git-branch-stage-commit.md

```diff
@@ -0,0 +1,120 @@
+# Git Basics (branch, stage, commit)
+Suppose you have just forked and cloned a repository onto your local machine. You're interested in making some changes to it, then eventually requesting that your changes be merged back into that repository. How do you get there?
+
+A good place to start is always to check the state of your files.
+
+Make sure you're in the directory that you cloned locally, then type in the command: 
+`git status`. 
+You'll see something like this:
+```
+On branch master
+Your branch is up-to-date with 'origin/master'.
+nothing to commit, working tree clean
+```
+
+Ok... cool. What does that mean? 
+
+## Branching
+### What is branching?
+When you branch, you diverge from the main line of development so that you can work on some changes without affecting the main line. 
+
+### When do I branch?
+You'll typically create new branches when you're developing a feature or putting in a fix. Keeping your changes organized and separated into branches will allow you to work faster overall and with less complications. 
+
+For example, let's say your project's main branch is `master` and it is the 'source of truth' of the state of your live app- everything that is merged into this branch should be ready to be online. Now let's say there are two new features you want to add to your project. If the features are not dependent on each other, this is a good opportunity to create a new branch for each of them. 
+
+Why not make one new branch and work on both of them there? Here's one reason: what if you do, but after you've started making changes for both features, you decide that you don't want feature #1 anymore? Now you have some unwanted changes scattered among your files, and you have to spend time finding and undoing those changes without messing with any of the changes that are related to feature #2. If you had just separated the features into their own branches, you could just scrap the branch for feature #1 and be done with it! 
```

Comment:
May want to add that keeping them separate, makes it easier to get your code reviewed also - smaller packages of code in a PR to review makes it easier for reviewers, etc.

Reply:
@codebyuma good point! added